### PR TITLE
Use catalog from network-of-terms-catalog

### LINKED ISSUE #5: Cache catalog and queries on server start
Currently, the `catalog.ttl` and `.rq` query files are read on each request. We can optimise this by reading and caching them on server start.

### PR DESCRIPTION
**Pull request by ddeboer**
Fix #5.

### CODE REVIEW

File: src/commands/sources/list.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
-import {CatalogService, Distribution} from '../../services/catalog';
 import {cli} from 'cli-ux';
 import {Command, flags} from '@oclif/command';
-import * as Logger from '../../helpers/logger';
+import {
+  Catalog,
+  Dataset,
+} from '@netwerk-digitaal-erfgoed/network-of-terms-catalog';
 
 export class ListSourcesCommand extends Command {
```
Comment (sdevalk) [inline src/commands/sources/list.ts:8]: The `flags` aren't used anymore. Remove these?

File: src/services/catalog.ts [hunk 2]
```diff
@@ -1,167 +0,0 @@
-import {ActorInitSparql} from '@comunica/actor-init-sparql/lib/ActorInitSparql-browser';
```
Comment (sdevalk) [inline src/services/catalog.ts:1]: This file - `catalog.ts` - still exists, though empty. Remove it?

File: src/services/distributions.ts [hunk 3]
```diff
@@ -1,71 +1,70 @@
...
-    const accessService = await this.catalogService.getAccessServiceByDistributionId(
-      args.distributionId
-    );
-    if (accessService === null) {
+    this.logger.info(`Preparing to query source "${args.source}"...`);
+    const dataset = await this.catalog.getByIdentifier(args.source);
+    if (dataset === undefined) {
       throw Error(
-        `Access service of source "${args.distributionId}" not found in catalog`
+        `Dataset with identifier "${args.identifier}" not found in catalog`
```
Comment (sdevalk) [inline src/services/distributions.ts:52]: This throws the error underneath if a non-existing source is given:

`Dataset with identifier "undefined" not found in catalog`

Also, perhaps change the word `Dataset` to `Source`?

File: src/services/query.ts [hunk 4]
```diff
@@ -65,29 +65,32 @@ export class QueryTermsService {
...
-      this.accessService.query,
+      this.dataset.distribution.query,
       config
     )) as IActorQueryOperationOutputQuads;
 
     return new Promise((resolve, reject) => {
       const termsTransformer = new TermsTransformer();
-      result.quadStream.on('error', reject);
+      result.quadStream.on('error', error => {
+        console.log('error', error);
```
Comment (sdevalk) [inline src/services/query.ts:80]: Is it necessary to log the error here? Can we leave it to the caller and simply reject instead?

File: src/services/terms.ts [hunk 5]
```diff
@@ -32,7 +32,7 @@ export class TermsTransformer {
 
   fromQuad(quad: RDF.Quad): void {
     const subject = quad.subject;
-    if (!subject.equals(this.currentTerm.id)) {
+    if (this.currentTerm.id === undefined || !subject.equals(this.currentTerm.id)) {
```
Comment (sdevalk) [inline src/services/terms.ts:35]: The new condition isn't really necessary: if `this.currentTerm.id` is undefined, `subject.equals()` returns true?

Comment (ddeboer) [reply src/services/terms.ts:35]: This caused `gts check` violation before.

Comment (ddeboer) [reply src/services/terms.ts:35]: Just

```
if (!subject.equals(this.currentTerm.id)) {
```

gives:

```
Error:(35, 25) TS2345: Argument of type 'Term | undefined' is not assignable to parameter of type 'Term'.
  Type 'undefined' is not assignable to type 'Term'.
```

**Comment by ddeboer** [comment]
Thanks for your feedback! Most issues are resolved now.

File: src/services/terms.ts (hunk 5 above)
Comment (sdevalk) [reply src/services/terms.ts]: Good one! :)

(Somewhat strange, though - the [RDFJS spec](https://rdf.js.org/data-model-spec/#term-interface) states that `equals()` returns `true` if (among others) the parameter is not `undefined`, so I expected `undefined` to be a valid value.)

**Pull request merged**